src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleRegistrationPhpGenerator.java

```diff
@@ -28,8 +28,12 @@ public ModuleRegistrationPhpGenerator(@NotNull ModuleRegistrationPhpData moduleR
     }
 
     public PsiFile generate(String actionName) {
-        ModuleDirectoriesData moduleDirectoriesData = directoryGenerator.createOrFindModuleDirectories(moduleRegistrationPhpData.getPackageName(), moduleRegistrationPhpData.getModuleName(), moduleRegistrationPhpData.getBaseDir());
-        return fileFromTemplateGenerator.generate(RegistrationPhp.getInstance(), getAttributes(), moduleDirectoriesData.getModuleDirectory(), actionName);
+        if (moduleRegistrationPhpData.getCreateModuleDirs()) {
+            ModuleDirectoriesData moduleDirectoriesData = directoryGenerator.createOrFindModuleDirectories(moduleRegistrationPhpData.getPackageName(), moduleRegistrationPhpData.getModuleName(), moduleRegistrationPhpData.getBaseDir());
+            return fileFromTemplateGenerator.generate(RegistrationPhp.getInstance(), getAttributes(), moduleDirectoriesData.getModuleDirectory(), actionName);
+        } else {
+            return fileFromTemplateGenerator.generate(RegistrationPhp.getInstance(), getAttributes(), moduleRegistrationPhpData.getBaseDir(), actionName);
```

Comment:
Same here, we don't need the `else` ramification.

Reply:
✅